staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -269,6 +269,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 func (s *WatchServer) HandleWS(ws *websocket.Conn) {
 	defer ws.Close()
 	done := make(chan struct{})
+	// ensure the connection times out
```

Comment:
This part looks fine - it looks like omission in handling websockets...

Reply:
agreed, would be nice if a test exercised this